### PR TITLE
📖  Instructions to connect to Grafana Pyroscope graphs and APF Dashboard

### PR DESCRIPTION
**Pull request by dumb0002**
## Summary

This PR adds instructions on how to connect to the following graphs to monitor KS control plane:   

-  API Server Priority & Fairness dashboard: for API Server in WDS, ITS spaces as well as the hosting cluster
-  Pyroscope profiling graphs: for KS controllers in the WDS & ITS spaces

## Related issue(s)

Fixes #

### CODE REVIEW

File: monitoring/README.md [hunk 1]
```diff
@@ -6,112 +6,135 @@
...
+b) API Server Priority & Fairness: input into the following text boxes.
+- `NS`: name of the KubeStellar space namespace, e.g., *wds1-system*, *wds2-system*, *its1-system*, etc.
+- `APIService`: name of the service for a KubeStellar space APIserver, e.g., *wds1*, *wds2*, *vcluster*, etc.
 
-    To monitor a WDS space (e.g., wds1) use the following parameters: 
-    - `SpaceNS: wds1-system` and `KSController: kubestellar-controller-manager-metrics-service`
-    - `SpaceNS: wds1-system` and `KSController: ks-transport-controller`
+    <img src="images/apiserver-apf-monitoring.png" width="60%" height="80%" title="apiserver-apf-monitoring.png"> 
+
+Note: use the same combinatios of `NS` and `APIService` parameters described in `step 5-a` to monitor the APIserver in the WDS space, ITS space or the hosting cluster.
```
[the remark this answers is not in the document]
Comment (dumb0002) [reply monitoring/README.md]: Fixed! thanks

File: monitoring/README.md [hunk 2]
```diff
@@ -6,112 +6,135 @@
...
+If you have more than one ITS, you can run the above script with the `--space-name` flag. For example: `./configure-metrics-its.sh --space-name its2`
 
+Optionally, check the prometheus service monitor objects for the ITS space:
 
-3. Connect to Grafana UI: 
+```bash
+$ kubectl -n ks-monitoring get servicemonitor -l "app.kubernetes.io/part-of!=kube-prometheus-stack"
+NAME                           AGE
+its1-apiserver                 64m
+its1-status-addon-controller   64m
```
Comment (MikeSpreitzer) [inline monitoring/README.md:64]: This is still smaller than the listing I get:

```shell
mspreitz@mjs13 monitoring % kubectl -n ks-monitoring get servicemonitor -l "app.kubernetes.io/part-of!=kube-prometheus-stack"  
NAME                           AGE
its1-apiserver                 10s
its1-status-addon-controller   10s
wds1-apiserver                 27s
wds1-ks-controller-manager     29s
wds1-ks-transport-controller   28s
```

File: monitoring/README.md [hunk 3]
```diff
@@ -6,112 +6,135 @@
...
+#### 6. View Pyroscope profile graphs for KubeStellar controllers in Grafana: 
+
+a) Connect to Pyroscope dashboard: 
+    i) Click `Connections -> Data Sources` in the primary menu
+    ii) Click `Explore` in the Pyroscope datasource selection
+
+b) Query profiles for the KubeSterllar controllers: for each controller, input a profile query (see sample below) and then click the button `Run query` (on the top right).
+- kubestellar-controller-manager: query: `{service_name="wds1-system/manager"}`
+- transport-controller: query: `{service_name="wds1-system/transport-controller"}`
+- addon-status-controller: query: `{service_name="its1-system/status-controller"}`
```
Comment (MikeSpreitzer) [inline monitoring/README.md:137]: This one gets me nothing.
image

**Review by MikeSpreitzer — COMMENTED** [review]
I have tested this and left some individual comments.

File: monitoring/README.md (hunk 2 above)
Comment (dumb0002) [reply monitoring/README.md:62]: Fixed!

File: monitoring/README.md (hunk 3 above)
Comment (dumb0002) [reply monitoring/README.md:135]: I'm aware of this issue - Pyroscope is not scraping the pod in the its1 space although all the configurations are set correctly - I am still looking deep into this and I will create a new PR once I figure out the bug.

**Comment by dumb0002** [comment]
> I have tested this and left some individual comments.

Thanks, I just replied to all your comments above.

File: monitoring/README.md (hunk 3 above)
Comment (dumb0002) [reply monitoring/README.md]: Fixed! Added a short message to the user.

**Comment by MikeSpreitzer** [comment]
FYI, regarding the last push:
image

The compare shows that it makes no change:

image

**Review by MikeSpreitzer — COMMENTED** [review]
/lgtm

**Pull request merged**